### ISSUE TITLE
Seems not to be working with RUT240 7.01.02

### ISSUE THREAD

**Issue by anhorbc** [meri-imperiumi/signalk-teltonika-rutx11#61]
Hi!
I get the following errors with RUT240. Any ideas?
Feb 18 20:54:00 POST /skServer/plugins/signalk-teltonika-rutx11/config 200 4.225 ms - 55
Feb 18 20:54:00 Could not parse JSON options:Unexpected end of JSON input
Feb 18 20:54:00 Could not parse JSON options:Unexpected end of JSON input
Feb 18 20:54:00 GET /skServer/plugins 200 29.120 ms - -
Feb 18 20:54:04 GET /skServer/debugKeys 304 1.586 ms - -
Feb 18 20:54:06 (node:473) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'message' of null at getData.then.then.then.then.then.catch (/home/pi/.signalk/node_modules/@meri-imperiumi/signalk-teltonika-rutx11/index.js:141:32)
Feb 18 20:54:06 (node:473) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
Feb 18 20:54:06 (node:473) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Feb 18 20:54:08 (node:473) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'message' of null at getData.then.then.then.then.then.catch (/home/pi/.signalk/node_modules/@meri-imperiumi/signalk-teltonika-rutx11/index.js:141:32)
Feb 18 20:54:08 (node:473) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
Feb 18 20:54:10 (node:473) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'message' of null at getData.then.then.then.then.then.catch (/home/pi/.signalk/node_modules/@meri-imperiumi/signalk-teltonika-rutx11/index.js:141:32)
Feb 18 20:54:10 (node:473) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
Feb 18 20:54:30 (node:473) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'message' of null at getData.then.then.then.then.then.catch (/home/pi/.signalk/node_modules/@meri-imperiumi/signalk-teltonika-rutx11/index.js:141:32)
Feb 18 20:54:30 (node:473) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
Feb 18 20:55:30 (node:473) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'message' of null at getData.then.then.then.then.then.catch (/home/pi/.signalk/node_modules/@meri-imperiumi/signalk-teltonika-rutx11/index.js:141:32)
Feb 18 20:55:30 (node:473) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 5)

**Comment by bergie** [comment]
Hmm… I don’t have a 240 to test with, maybe @KEGustafsson has an idea on this one?

**Comment by KEGustafsson** [comment]
I need to check are the modbus registers changed from fw 6.x to 7.x.

Fw 7.x was major upgrade, which changed  lots of things. Also I know that there are few hw revisions (chipsets) in RUT240, which are not sw wise compatible. All features in some revisions are not working or they are working badly.

Current RUT240 setup was done against 6.x firmware.

**Comment by KEGustafsson** [comment]
Feb 18 20:54:00 POST /skServer/plugins/signalk-teltonika-rutx11/config 200 4.225 ms - 55
Feb 18 20:54:00 Could not parse JSON options:Unexpected end of JSON input
Feb 18 20:54:00 Could not parse JSON options:Unexpected end of JSON input
Feb 18 20:54:00 GET /skServer/plugins

There is something weird here!?!
Your config file is somehow broken.
Can you just delete you current rutx11 config file and setup it again from Plugin Config.

**Comment by anhorbc** [comment]
The errors "Could not parse JSON options:Unexpected end of JSON input" came from other plugins.
I deleted the rutx11 config file and rconfigured it. I still get the following errors: "Feb 24 22:32:46 (node:32123) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'message' of null at getData.then.then.then.then.then.catch ***@***.***/signalk-teltonika-rutx11/index.js:141:32)
Feb 24 22:32:46 (node:32123) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
Feb 24 22:33:46 (node:32123) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'message' of null at getData.then.then.then.then.then.catch ***@***.***/signalk-teltonika-rutx11/index.js:141:32)
Feb 24 22:33:46 (node:32123) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)

Have I configured the modbus client package in the teltonika router correctly? I am not very familiar with modbus.

/A

Hämta Outlook för Android<https://aka.ms/AAb9ysg>
